src/index.js

```diff
@@ -1,42 +1,57 @@
 // ITERATION 1
 
+
+
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span");
+  let quantity = product.querySelector(".quantity input");
+  let subtotal = product.querySelector(".subtotal span");
+  let total = price.innerHTML * quantity.value;
+  return subtotal.innerHTML = total; 
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+function calculateAll() {
 
-  // ITERATION 3
-  //... your code goes here
+  let products = document.getElementsByClassName('product');
+  let sumOfProducts = 0;
+  for(let i = 0; i < products.length ; i++) {
+    sumOfProducts += updateSubtotal(products[i]);
+  }
+  
+  let finalTotal = document.querySelector("#total-value span");
+  finalTotal.innerHTML = sumOfProducts;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  console.log('The target in remove is:', target.parentNode.parentNode);
+  target.parentNode.parentNode.remove();
+  calculateAll();
 }
 
 // ITERATION 5
+// couldn´t figure out this one yet :/
 
 function createProduct() {
-  //... your code goes here
+  let newProduct = document.createElement(".create-product input");
```

Comment:
If you're using `document.createElement()` what you need to pass inside the parentheses is the specific html tag that you want to create so in this case it would be something like `document.createElement('tr')`